storage/database/s3filedb.go

```diff
@@ -0,0 +1,148 @@
+package database
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/klaytn/klaytn/common/hexutil"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/klaytn/klaytn/log"
+)
+
+// s3FileDB is an implementation of fileDB based on AWS S3.
+// It stores the data to the designated AWS S3 bucket.
+type s3FileDB struct {
+	region   string
+	endpoint string
+	bucket   string
+	s3       *s3.S3
+	logger   log.Logger
+}
+
+// newS3FileDB returns a new s3FileDB with the given region, endpoint and bucketName.
+// If the given bucket does not exist, it creates one.
+func newS3FileDB(region, endpoint, bucketName string) (*s3FileDB, error) {
+	localLogger := logger.NewWith("endpoint", endpoint, "bucketName", bucketName)
+	sessionConf, err := session.NewSession(&aws.Config{
+		Region:           aws.String(region),
+		Endpoint:         aws.String(endpoint),
+		S3ForcePathStyle: aws.Bool(true),
+	})
+	if err != nil {
+		localLogger.Error("failed to create session", "region", region)
+		return nil, err
+	}
+
+	s3DB := &s3FileDB{
+		region:   region,
+		endpoint: endpoint,
+		bucket:   bucketName,
+		s3:       s3.New(sessionConf),
+		logger:   localLogger,
+	}
+
+	exist, err := s3DB.hasBucket(bucketName)
+	if err != nil {
+		localLogger.Error("failed to retrieve a bucket list", "err", err)
+		return nil, err
+	}
+	if !exist {
+		localLogger.Warn("cannot find the bucket. creating the bucket")
+		_, err = s3DB.s3.CreateBucket(&s3.CreateBucketInput{
+			Bucket: aws.String(bucketName),
+		})
+		if err != nil {
+			localLogger.Error("failed to create a bucket", "err", err)
+			return nil, err
+		}
+	}
+	return s3DB, nil
+}
+
+// hasBucket returns if the bucket exists in the endpoint of s3FileDB.
+func (s3DB *s3FileDB) hasBucket(bucketName string) (bool, error) {
+	output, err := s3DB.s3.ListBuckets(&s3.ListBucketsInput{})
+	if err != nil {
```

Comment:
```suggestion
	// TODO-Klaytn : handle with `awserr.Error` type
	if err != nil {
```
When the bucket name doesn't meet the restriction, an error occurs without a log.

Reply:
@winnie-byun What does "an error occurs without a log" mean?

Reply:
I figured my problem might not be relevant to this code and I need to take a look.
When I run a test, the process fails with a call stack and does not contain an error log.

Reply:
okay. let me know when you figure it out.

Reply:
Found out other problem